integreat_cms/cms/views/settings/totp_register_view.py

```diff
@@ -0,0 +1,93 @@
+import base64
+import logging
+import pyotp
+from django.views.generic import TemplateView
+from django.shortcuts import render, redirect
+from django.contrib import messages
+from django.utils.translation import ugettext_lazy as _
+from django.utils.decorators import method_decorator
+
+from ...decorators import modify_mfa_authenticated
+from ...utils.totp_utils import generate_totp_qrcode, check_totp_code
+
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(modify_mfa_authenticated, name="dispatch")
+class TOTPRegisterView(TemplateView):
+    """
+    View to register a TOTP authenticator.
+    """
+
+    template_name = "authentication/register_totp.html"
+
+    def get(self, request, *args, **kwargs):
+        r"""
+        Initialize the registration for the authentication application.
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :return: A redirection to the account settings
+        :rtype: ~django.http.HttpResponseRedirect
+        """
+
```

Comment:
```suggestion
        if request.user.totp_key:
            messages.error(
                request,
                _(
                    "You have already registered a TOTP key. Please disconnect your app before adding a new one."
                ),
            )
            kwargs = {"region_slug": request.region.slug} if request.region else {}
            return redirect("user_settings", **kwargs)
```
I guess we could make sure a key is not already configured to prevent accidental overwriting of existing keys...